docs/application/native/guides/compliations/watchface-complication.md

```diff
@@ -0,0 +1,329 @@
+		const char *provider_id,
+		watchface_complication_type_e type,
+		watchface_complication_error_e error,
+		void *user_data)
+{
+}
+
+static void _init_complication()
+{
+	watchface_complication_add_updated_cb(complication,
+		_watchface_complication_updated_cb,
+		_watchface_complication_error_cb,  NULL);
+} 
+```
+ 
+When the complication provider data is updated, `watchface_complication_updated_cb()` is called.
+Watch Face can get the information from the bundle parameter of callback by using `watchface_complication_data_get_*`.
+
+To get the complication provider data, the following functions can be used:
+
+
+| Function name | Data type |
+|---------------|-----------|
+| `watchface_complication_data_get_type()` | watchface complication type |
+| `watchface_complication_data_get_short_text()` | short text |
+| `watchface_complication_data_get_long_text()` | long text |
+| `watchface_complication_data_get_title()` | title text |
+| `watchface_complication_data_get_timestamp()` | timestamp |
+| `watchface_complication_data_get_image_path()` | image path |
+| `watchface_complication_data_get_ranged_value()` | min / max / current value |
+| `watchface_complication_data_get_icon_path()` | icon path |
+| `watchface_complication_data_get_extra_data()` | extra data |
+
+```cpp
+void _watchface_complication_updated_cb(int complication_id,
+		const char *provider_id,
+		watchface_complication_type_e type,
+		const bundle *data, void *user_data)
+{
+	char *shorttext;
+	if (type == WATCHFACE_COMPLICATION_TYPE_SHORT_TEXT)
+		watchface_complication_data_get_short_text(data, &shorttext);
+}
+```
+
+
+If the callback is not used any more, it must be removed by using `watchface_complication_remove_updated_cb()`:
+
+```cpp
+{
+	watchface_complication_remove_updated_cb(complication,
+		_watchface_complication_updated_cb);
+}
+```
+
+
+If the watch face wants the updated complication data, watch face can request to update the data to the complication provider:
+
+```cpp
+{
+	watchface_complication_send_update_request(complication);
+}
+```
+
+
+## Specifying to Support only Specific Complication Providers
+
+Watch Face can support the specific complication provider by using `allowed_list`.
+In This case, watch face supports complications only in allowed complication list:
+
+```cpp
+{
+	complication_allowed_list_h allowed_list;
+
+	watchface_complication_allowed_list_create(&allowed_list); 
+	watchface_complication_allowed_list_add(allowed_list, "PROVIDER_ID_1", 
+			WATCHFACE_COMPLICATION_TYPE_SHORT_TEXT); 
+	watchface_complication_allowed_list_add(allowed_list, "PROVIDER_ID_2", 
+			WATCHFACE_COMPLICATION_TYPE_SHORT_TEXT); 
+	watchface_complication_allowed_list_apply(complication, allowed_list);
+
+	watchface_complication_allowed_list_destroy(allowed_list);
+}
+```
+
+
+## Watch Face Edit with Editable Feature
+
+The watch face application can provide the changeable features (such as, complication, color, font and so on) with watch face editable APIs.

```

Comment:
The watch face application can provide the changeable features (such as, complication, color, font, and so on) with watch face editable APIs.